### PR TITLE
fix tls with jwt authn usecase

### PR DESCRIPTION
**Pull request by tanujd11**
**What type of PR is this?**

Bug Fix

**What this PR does / why we need it**:

HTTPS and JWT authentication was not working in tandem because for some reason we were getting the information for JWTRequirement from HCM in DefaultFilterChain. DefaultFilterChain is not populated in case of HTTPS.

The question is irRoute already have all the information about JWT Requirement as the HCM of listener is also populated from route only which only checks if the listener has JWT Filter on the route and patch listener. Then why reverse check is done if httpfilter has the JWT requirements?  

**Which issue(s) this PR fixes**:

Fixes https://github.com/envoyproxy/gateway/issues/1768

### CODE REVIEW

**Review by Xunzhuo — APPROVED** [review]

**Comment by zirain** [comment]
can you add test case for this?

**Comment by tanujd11** [comment]
@zirain somehow `make testdata` generated test file is failing in `go.test.coverage`. Default `make test` is working. This is due to some unicode sequence. I am not sure why is this happening with this PR only. Any ideas?

**Comment by zirain** [comment]
seems related to https://github.com/envoyproxy/gateway/pull/1859

File: internal/xds/translator/authentication.go [hunk 1]
```diff
@@ -195,29 +194,23 @@ func buildXdsUpstreamTLSSocket() (*corev3.TransportSocket, error) {
 }
 
 // patchRouteWithJwtConfig patches the provided route with a JWT PerRouteConfig, if the
-// route doesn't contain it. The listener is used to create the PerRouteConfig JWT
-// requirement.
-func patchRouteWithJwtConfig(route *routev3.Route, irRoute *ir.HTTPRoute, listener *listenerv3.Listener) error { //nolint:unparam
+// route doesn't contain it.
+func patchRouteWithJwtConfig(route *routev3.Route, irRoute *ir.HTTPRoute) error { //nolint:unparam
```
Comment (zirain) [inline internal/xds/translator/authentication.go:198]: why `unparam`?

Comment (tanujd11) [reply internal/xds/translator/authentication.go]: Have removed it. It was there earlier so somehow I missed that. Thanks for pointing out

**Review by arkodg — APPROVED** [review]
LGTM, thanks !

**Review by zirain — APPROVED** [review]

**Review by arkodg — APPROVED** [review]

**Pull request merged**